tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -515,73 +531,88 @@ func (s *BCDConsumerIntegrationTestSuite) Test7ConsumerFPRewards() {
 
 // Test8BabylonFPCascadedSlashing
 // 1. Submits a Babylon FP valid finality sig to Babylon
-// 2. Block is finalized.
-// 3. Equivocates/ Submits a invalid finality sig to Babylon
+// 2. Block is finalized (but not yet timestamped)
+// 3. Equivocates/ Submits an invalid finality sig to Babylon
 // 4. Babylon FP is slashed
-// 5. Babylon notifies involved consumer about the delegations.
-// 6. Consumer discounts the voting power of other involved consumer FP's in the affected delegations
+// 5. Babylon notifies the involved Consumer about the delegations.
+// 6. Consumer discounts the voting power of other involved consumer FPs in the affected delegations
 func (s *BCDConsumerIntegrationTestSuite) Test8BabylonFPCascadedSlashing() {
 	// get the activated height
 	activatedHeight, err := s.babylonController.QueryActivatedHeight()
 	s.NoError(err)
 	s.NotNil(activatedHeight)
-
-	// get the block at the activated height
-	activatedHeightBlock, err := s.babylonController.QueryCometBlock(activatedHeight.Height)
-	s.NoError(err)
-	s.NotNil(activatedHeightBlock)
+	submitHeight := activatedHeight.Height
 
 	// get the babylon finality provider
 	babylonFp, err := s.babylonController.QueryFinalityProviders()
 	s.NoError(err)
 	s.NotNil(babylonFp)
-
 	babylonFpBIP340PK := bbn.NewBIP340PubKeyFromBTCPK(babylonFpBTCPK)
-	randIdx := activatedHeight.Height - 1 // pub rand was committed from height 1-100
-
-	// submit finality signature
-	txResp, err := s.babylonController.SubmitFinalitySignature(
-		babylonFpBTCSK,
-		babylonFpBIP340PK,
-		randListInfo1.SRList[randIdx],
-		&randListInfo1.PRList[randIdx],
-		randListInfo1.ProofList[randIdx].ToProto(),
-		activatedHeight.Height)
-	s.NoError(err)
-	s.NotNil(txResp)
 
-	// ensure vote is eventually cast
-	var votes []bbn.BIP340PubKey
-	s.Eventually(func() bool {
-		votes, err = s.babylonController.QueryVotesAtHeight(activatedHeight.Height)
-		if err != nil {
-			s.T().Logf("Error querying votes: %v", err)
-			return false
-		}
-		return len(votes) > 0
-	}, time.Minute, time.Second*5)
-	s.Equal(1, len(votes))
-	s.Equal(votes[0].MarshalHex(), babylonFpBIP340PK.MarshalHex())
+	// Eventually find a block that is not timestamped
+	for {
+		s.T().Logf("Trying to equivocate for the block at height: %d", submitHeight)
```

Comment:
Hmm will this wait for too long?

Reply:
This could be improved by sending the equivocation for a not-yet-finalised block.

This is fast anyway, as the finalisation process is relatively straightforward.

Reply:
I see. So after the first equivocation the loop should break. Is this for loop necessary here then? For example, how about we do a first equivocation and the correpsonding asserts, and then try to submit finality again and expect error?

Reply:
I think it's necessary, because we can only know about timestamped (Babylon) epochs, and not timestamped (BSN) heights.

In fact, this of course needs more thinking, but it could be a good idea to add a `timestamped` boolean to the block information, in the same way we add a `finalised` one.